cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredRow.java

```diff
@@ -81,7 +102,23 @@ public Double getDouble(String fieldName) throws InvalidFieldException {
   @Nullable
   @Override
   public byte[] getBytes(String fieldName) throws InvalidFieldException {
-    return isNull(fieldName) ? null : struct.getBytes(fieldName).toByteArray();
+    if (isNull(fieldName)) {
+      return null;
+    }
+
+    byte[] value = struct.getBytes(fieldName).toByteArray();
+    CompressorType compressorType = getCompressorType(fieldName);
+    if (value == null || compressorType == null) {
+      return value;
+    }
+
+    try {
+      Compressor compressor = CompressorFactory.getCompressor(compressorType);
+      value = compressor.decompress(value);
+    } catch (IOException e) {
+      Throwables.propagate(e);
```

Comment:
do `throw Throwables.propagate(e)`. It's better from java compilation perspective, see javadoc. Ditto in other cases